stl/inc/any

```diff
@@ -178,7 +177,7 @@ public:
                                     int> = 0>
     any& operator=(_ValueType&& _Value) {
         // replace contained value with an object of type decay_t<_ValueType> initialized from _Value
-        *this = any{_STD forward<_ValueType>(_Value)};
+        _Assign(any{_STD forward<_ValueType>(_Value)});

```

Comment:
This change will hopefully not affect performance of `operator=(const&)` and `operator=(auto&&)`.
For `operator=(any&&)`, it seems for correctness we have to make a local variable as well.